lms/djangoapps/courseware/field_overrides.py

```diff
@@ -70,6 +69,12 @@ def wrap(cls, user, wrapped):
         any performance impact of this feature if no override providers are
         configured.
         """
+        # if ccx not enabled for this course, just return unwrapped
+        from courseware.courses import get_course_by_id
+        course = get_course_by_id(course_id)
+        if not course.enable_ccx:
+            return wrapped
```

Comment:
Rather than loading the course XBlock here, it would be better to pass the instantiated course in. (As is, you're going to encur additional database access).